frame/ibc/src/lib.rs

```diff
@@ -450,7 +451,7 @@ pub mod pallet {
 				counterparty,
 				version: Some(version),
 				delay_period,
-				signer: Signer::new(""),
+				signer: Signer::from_str("pallet_ibc").map_err(|_| Error::<T>::DecodingError)?,
```

Comment:
Can we make Signer a const to avoid doing the work during runtime; and reducing the error noise?

Reply:
The `Signer` struct holds a `String`, so  it can't be represented as a constant.

Reply:
:( Badly designed APIs lol